### ISSUE TITLE
dfx.json type azle

### ISSUE THREAD

**Issue by lastmjs** [dfinity/sdk#2528]
It would be great to add official [Azle](https://github.com/demergent-labs/azle) support to `dfx.json` with `"type": "azle"`. This would greatly simplify the current `dfx.json` configuration necessary for Azle applications, which looks like this:

```json
{
    "canisters": {
        "simple_erc20": {
            "type": "custom",
            "build": "npx azle simple_erc20",
            "root": "src",
            "ts": "src/simple_erc20.ts",
            "candid": "src/simple_erc20.did",
            "wasm": "target/wasm32-unknown-unknown/release/simple_erc20.wasm",
            "declarations": {
                "output": "test/dfx_generated/simple_erc20"
            }
        }
    }
}
```

**Comment by sesi200** [comment]
Hi Joardan, generally we'd like to support this, but we have a lot of open questions. @dfx-json would like to talk to you about this and has tried to reach you on Discord. Would you be willing to talk to him?

**Comment by lastmjs** [comment]
Yes that would be great, I don't see a message on Discord by I'm @lastmjs basically everywhere

**Comment by domwoe** [comment]
@sesi200 Do we have an update on the plugin system to support other cdks?

**Comment by sesi200** [comment]
@smallstepman is working on it, but he's on PTO for this and next week AFAIK.
